src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -262,7 +283,22 @@ public function isDeprecated()
      */
     public function getDeprecationMessage(string $node, string $path)
     {
-        return strtr($this->deprecationMessage, ['%node%' => $node, '%path%' => $path]);
+        trigger_deprecation('symfony/config', '5.1', 'The "%s()" method is deprecated, use "getDeprecation()" instead.', __METHOD__);
+
+        return $this->getDeprecation($node, $path)['message'];
+    }
+
+    /**
+     * @param string $node the configuration node name
+     * @param string $path the path of the node
+     */
+    public function getDeprecation(string $node, string $path)
```

Comment:
this method currently triggers notices if you call it on a non-deprecated node, as `$this->deprecation` won't have these keys.

Reply:
I remplaced `$this->deprecation[..]` by `$this->deprecation[..] ?? ''`